brushfire-scalding/src/main/scala/com/stripe/brushfire/scalding/Trainer.scala

```diff
@@ -185,6 +185,39 @@ case class Trainer[K: Ordering, V, T: Monoid](
   }
 
   /**
+   * prune a tree to minimize validation error
+   *
+   * Construct a Map[Int,T] from the trainingData for each tree, and then transform the trees using the prune method.
+   *
+   */
+  def prune[P, E](path: String, error: Error[T, P, E])(implicit voter: Voter[T, P], inj: Injection[Tree[K, V, T], String], ord: Ordering[E]): Trainer[K, V, T] = {
+    flatMapTrees {
+      case (trainingData, sampler, trees) =>
+        lazy val treeMap = trees.toMap
+
+        val newEx = trainingData
+          .flatMap { instance =>
+            for { // Iterate over any trees for which this instance is a validation instance.
+              (treeIndex, tree) <- treeMap if sampler.includeInValidationSet(instance.id, instance.timestamp, treeIndex);
+              leafIndex <- tree.leafIndexFor(instance.features).toList // Find the leaf that this instance falls into.
+            } yield treeIndex -> Map(leafIndex -> instance.target) // Yield validation instances for each tree, leaf.
+          }
+          .sumByKey // Make a single Map for each treeIndex. The Map is from leafIndex to validation target.
+          .withReducers(reducers)
+          .mapGroup {
```

Comment:
`UnsortedGrouped` doesn't have a `map` method as far as I can tell, but I can do `.toTypedPipe.map`. Does that make sense?